staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -117,25 +117,63 @@ type DiffOptions struct {
 	pruner           *pruner
 }
 
-func NewDiffOptions(ioStreams genericclioptions.IOStreams) *DiffOptions {
-	return &DiffOptions{
+// DiffFlags directly reflect the information that CLI is gathering via flags.  They will be converted to Options, which
+// reflect the runtime requirements for the command.  This structure reduces the transformation to wiring and makes
+// the logic itself easy to unit test
+type DiffFlags struct {
+	FilenameOptions resource.FilenameOptions
+
+	ServerSideApply   bool
+	FieldManager      string
+	ForceConflicts    bool
+	ShowManagedFields bool
+
+	Selector string
+	Diff     *DiffProgram
```

Comment:
Why `Diff` is in `DiffFlags`?

Reply:
Doesn't make much sense. 
Thank you for the fast feedback. I sent another commit fixing the points you mentioned :)